Jenkinsfile_CNP

```
@@ -1,5 +1,11 @@
 #!groovy
-
+properties([
+    parameters([
+        booleanParam(name: 'SKIP_TESTS', defaultValue: false, description: 'Skip all tests'),
+        booleanParam(name: 'SKIP_SONAR', defaultValue: false, description: 'Skip sonar'),
+        booleanParam(name: 'FAIL_FAST', defaultValue: true, description: 'Should tests fail fast'),
+    ])
+])
```

I don't think we should add this, it could easily be mis-used.

---

Why do we want to skip sonar? And what does the fail fast one do?

---

FAST_FAIL will stop running fail the test run as soon as the first test fails instead of after running them all.
Essentially meaning we get easier feedback.

Sonar just saves us a few mins

---

I can sort of see the value in the skip tests one if you wanted to quickly try out a change on the Dev environment. But can't see the value in the other 2. Surely if you've got failing tests you would want to know all of the ones that are failing instead of just one. And Sonar issues should be quick to resolve anyway. @davet1985 @hemantasharma1129 any thoughts? I'd suggest removing `SKIP_SONAR` and `FAIL_FAST` from this config.

---

Sonar can be a pain like the tests if you want to get something out to the dev env quickly. I had a chat with @Ben-Edwards-cgi and we agreed that you wouldn't want to allow the build to pass if you are skipping sonar or tests.

Maybe we should set the `FAIL_FAST` default to false, to continue running as we are now, but allow the option?

---

I can see both ways,

I like FAIL_FAST being true just so the build fails as soon as it finds an error so I know I need to fix the code instead of having to wait the full 40 mins to know

On the other hand knowing the full set of test failures up front is handy to know so we can fix them all in one go.

I personally prefer fail fast so flakey tests are found earlier so a re-build can happen quicker, and if one integration test fails I usually run full integration locally to view the logs easier anyway.

But I am easy either way so happy to update the default to either

---

I think we keep it the same right now ie. default to false. @jackmaloney?

---

yes please

---

Cool updated :)